### PR TITLE
Add single column functionality to the Columns block

### LINKED ISSUE #24055: Columns Block: Allow to Have One Column
I often need to make a design like this example where a heading spans across the wide width.

Frame 1

But adding a Heading block straight would result like this.

Frame 2

This is because the Heading block doesn’t have the width setting like other blocks at the moment. So I need to use the column block that is set to the wide width. And the minimum column number in the block is 2, so what I need to do in the end is to set one of the columns to `99%` width, and the other to `1%`  so that the larger column would take *most* of the wide width.

Frame 3

While this gets the job kind of done, it’s really not ideal. Also, we see the empty column hanging around in the editor, and that's confusing for people who are editing the page. (I think I’ve seen this method was used in one of the block patterns in Gutenberg that had a large text saying “it’s time” at one point.)

There might be a better workaround or Gutenberg has already a better plan to accommodate this need, but otherwise, it could be a welcome addition to the block.

### PR DESCRIPTION
**Pull request by richtabor**
## Description
This PR does two things: 
1. Changes the min value of the `count` control for columns count to `1` from `2`. 
2. Adds a new one column variation.

This closes #24055, which will make building more robust column-based layouts easier. Instead of relying on switching between using Columns and Group blocks, single column functionality within a Columns block makes building interesting layouts easier and faster. For one, all blocks within a Group block can't always be aligned fullwidth/wide (Headings and Paragraphs for example), which result in a frustrating experience trying to better align content within columns.

## How has this been tested?
Tested in WP 5.5 RC.

### Min value set to 1:
Screen Shot 2020-07-20 at 12 51 14 PM

### Using the one column variation in practice:
Screen Shot 2020-07-20 at 12 50 31 PM

### New one column variation:
Screen Shot 2020-07-20 at 12 50 36 PM

## Types of changes
New feature:
1. Changes min prop of the `count` attribute for the Columns block
2. Adds a new one column variation

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by talldan** [comment]
It definitely seems like the alignments mentioned in #24055 is something that should be supported.

I wonder if columns is the right place to add this, though.

There's also Group. It doesn't work at the moment because of the way group constrains it's inner blocks using the `wp-block-group__inner-container` div, but I recall seeing it mentioned elsewhere that the inner container isn't ideal for full site editing where it'd be nice to use the Group as a more general container - https://github.com/WordPress/gutenberg/issues/20376.

Maybe the inner container should only be present when the user wants to constrain the inner blocks to the content width?

**Comment by richtabor** [comment]
> I wonder if columns is the right place to add this, though.

I think it is, for a few key reasons:

1. Columns are a relative unit of web design. It's contextual and familiar, whereas implementing Groups and Columns together is bit less meaningful. 

2. Having single column functionality would make it easier to build out pages, as sometimes its necessary to change a columns based layout around. Being able to simply "resize" a columns block down to one is a better experience than having to add a new Group block, then drag each block into that new block, then delete the old Columns block.

3. I think it would be much more confusing if the inner container of the Group block changes based on a setting/content width - than having one consistent experience on its grouped blocks.

**Comment by iamtakashi** [comment]
In my opinion, a single column option is a nice addition for a reason other than the lack of alignment option in the heading block. Showing an empty column isn't ideal for people who edit when the page only needs a single column.

screenshot

**Comment by iamtakashi** [comment]
It might be expected behaviour, but the single column is always the centre. So we'd still need to use an empty column to achieve something like the example above.

test

It feels out of scope for this PR to add the flexibility in position, but I just wanted to mention it.

**Comment by paaljoachim** [comment]
It would be helpful to have the option to be able to add 1 column in a layout.
Let's get this in..:)

**Comment by jasonlemay** [comment]
> Columns are a relative unit of web design. It's contextual and familiar, whereas implementing Groups and Columns together is bit less meaningful.

Couldn't agree more. I got stuck with the fact that single column wasn't available and had to resolve using a Group to group ... one thing!

**Comment by iamtakashi** [comment]
Another good thing to allow a single column is that it avoids a situation like below where columns are forced to be 50:50 for a medium size viewport.

Is there anything else we could do to get this in? Who can we ping for opinions?

File: packages/block-library/src/columns/variations.js [hunk 1]
```diff
@@ -12,6 +12,28 @@ import { __ } from '@wordpress/i18n';
...
+				xmlns="http://www.w3.org/2000/svg"
+			>
+				<Path
+					fillRule="evenodd"
+					clipRule="evenodd"
+					d="m39.0625 14h-30.0625v20.0938h30.0625zm-30.0625-2c-1.10457 0-2 .8954-2 2v20.0938c0 1.1045.89543 2 2 2h30.0625c1.1046 0 2-.8955 2-2v-20.0938c0-1.1046-.8954-2-2-2z"
+				/>
+			</SVG>
+		),
+		isDefault: true,
```
Comment (talldan) [inline packages/block-library/src/columns/variations.js:33]: I don't think there should be two variations with `isDefault: true`.

In practice this seems to get overridden by the `isDefault` on the two columns equal variation, so maybe lets remove this line.

File: packages/block-library/src/columns/variations.js [hunk 2]
```diff
@@ -12,6 +12,28 @@ import { __ } from '@wordpress/i18n';
  * @type {WPBlockVariation[]}
  */
 const variations = [
+	{
+		name: 'one-column-full',
+		title: __( '100' ),
+		description: __( 'One column; full width' ),
```
Comment (talldan) [inline packages/block-library/src/columns/variations.js:18]: This is open to interpretation, but 'full width' could end up being confused for the alignment option (especially for screenreader users). I think just leaving it at 'One column' would be fine, but there might also be phrasing like 'full size' or '100%' that could work.

**Review by talldan — COMMENTED** [review]
@richtabor I've reviewed the code changes. Just a couple of small comments.

I still wonder about the overlap with the group block, but I'm happy for those who have more experience building themes and content to make a decision on this.

**Comment by paaljoachim** [comment]
There are now two PRs that look to accomplish to adding a single column option.
https://github.com/WordPress/gutenberg/pull/25857

**Comment by retrofox** [comment]
> There are now two PRs that look to accomplish to adding a single column option.
> #25857

Closing ... Sorry I thought I did it before.

**Comment by paaljoachim** [comment]
It would be great if you can help @richtabor on this PR, and help get it merged. 
Thanks @retrofox Damien!

File: packages/block-library/src/columns/variations.js (hunk 2 above)
Comment (retrofox) [reply packages/block-library/src/columns/variations.js:18]:
```suggestion
		description: __( 'One column' ),
```

File: packages/block-library/src/columns/variations.js (hunk 1 above)
Comment (retrofox) [reply packages/block-library/src/columns/variations.js:33]:
```suggestion
```

we'd like two columns as default, do we?

**Comment by richtabor** [comment]
@paaljoachim Made the suggested changes; I think it's ready to go :)

File: packages/block-library/src/columns/variations.js (hunk 1 above)
Comment (richtabor) [reply packages/block-library/src/columns/variations.js]: Yup, I removed it. isDefault should be set to true for the two columns variation.

**Review by talldan — APPROVED** [review]
Looks good, thanks for your work!

**Comment by dianeco** [comment]
The One Column layout (100%) is only 50% on small screens because of the following CSS:
```
@media (min-width: 600px) and (max-width:781px) {
    .wp-block-column {
        flex-basis:calc(50% - 1em)!important;  
    }
}
```

**Pull request merged**